ext/standard/array.c

```diff
@@ -6052,7 +6052,8 @@ PHP_FUNCTION(array_key_exists)
 			RETVAL_BOOL(zend_hash_exists(ht, ZSTR_EMPTY_ALLOC()));
 			break;
 		case IS_DOUBLE:
-			RETVAL_BOOL(zend_hash_index_exists(ht, zend_dval_to_lval(Z_DVAL_P(key))));
+			/* TODO Should this warn or not? */
```

Comment:
As we also warn here for resource, and also throw the deprecation in isset(), I believe we so.

Reply:
Wouldn't it make sense to use a union type here? If I understand the behaviour correctly we would need `int|string|null` as `null` is evaluated as `''` instead of `0`, in which case we get the handling of the other types for free.

Reply:
As the conversion semantics are supposed to match those of normal array access, I think it's clearer not to use union types. If nothing else, we'd get a difference in error level for the resource case.

Reply:
ACK, will handle manually then.